### DISCUSSION TITLE
Refresh access token and reuse it for further iterations

### DISCUSSION THREAD

**Discussion by priya-dsp** [artilleryio/artillery#2715] [Artillery]
We are running a test for duration of more than one hour and have an API that generates Authorization bearer token which expires for an hour, how to rerun this api for every 45 minutes? All users generates unique token.

**Comment by bernardobridge** [comment]
Hi @priya-dsp,

Typically you would set that token in the context in a [`before` hook](https://www.artillery.io/docs/reference/test-script#before-and-after-sections), and then fetch a new token periodically using [e.g. `beforeScenario` hooks](https://www.artillery.io/docs/reference/engines/http#function-actions-and-beforescenario--afterscenario-hooks).

I'll add an example of doing this to our examples when I get a chance, as it's a common enough use case :)

**Reply by priya-dsp** [reply]
@bernardobridge ,Thanks for your reply. If we are using before hook to call Api that generates access token, then all Vusers will be using same access token. But, in our scenario, we need to generate access token for each Vusers and reuse the user token in following iterations. How to control the throughput of this API? Can you please provide an example?

**Reply by hassy** [reply]
hi @priya-dsp 👋 just added an example here: https://github.com/artilleryio/artillery/tree/main/examples/refresh-auth-token

let us know if you have any questions

**Reply by priya-dsp** [reply]
Sure, thank you @hassy
